map.js

```diff
@@ -195,143 +183,97 @@ function drawMapOverlay(){
         }
     }
     else if(currentOverlay == "NirnRoute"){
         for(const icon of overlay.nirnroots){
-            if(icon.contains(lastMouseLoc)){
-                //TODO
+            if(icon.contains(clickLocInMapSpace)){
+                console.log("nirnroot "+icon.cell.formId+" clicked");
                 return true;
             }
         }
     }
     return false;
 }
 
-/**
- * Construct a map location object from a location json cell.
- * @param {*} cell 
- */
-function MapIcon(cell){
-    MapObject.call(this);
-
-    this.cell = cell;
-
-    this.recalculateBoundingBox();
-    this.draw = drawIcon2;
-
-    if(cell.hive.classname == "nirnroot"){
-        this.icon = iconSwitch("Nirnroot");
-    }
-    else{
-        this.icon = iconSwitch(cell.icon);
-    }
-}
-MapIcon.prototype = Object.create(MapObject.prototype);
-
-/**
- * whenever we zoom, we will need to call this.
- */
-MapIcon.prototype.recalculateBoundingBox = function(){
-    let mapCoords = worldSpaceToMapSpace(this.cell.x, this.cell.y);
-    const halfHeightDown = Math.floor(iconH / 2);
-    const halfHeightUp = Math.ceil(iconH / 2);
-    this.minX = mapCoords.x - halfHeightDown;
-    this.minY = mapCoords.y - halfHeightDown;
-    //default state is only icon, so its easy
-    this.maxX = mapCoords.x + halfHeightUp;
-    this.maxY = mapCoords.y + halfHeightUp;
-}
-MapIcon.prototype.contains = function(point){
-    //since we store x and y in map space we gotta add that back in and uuuuuuugh
-    let mapPoint = new Point(point.x + mapOffset.x, point.y + mapOffset.y);
-    return Object.getPrototypeOf(Object.getPrototypeOf(this)).contains.call(this, mapPoint);
-}
-
-function drawIcon2(ctx){
-    //draws the name for the map icon if hovered.
-    if(this.cell.hive.classname != "nirnroot"){
-        if(this.contains(lastMouseLoc)){
-            //create rect that contains text and the icon
-            ctx.beginPath();
-            ctx.fillStyle = "#E5D9B9";
-            ctx.rect(this.minX-mapOffset.x, this.minY-mapOffset.y, (this.cell.name.length * 10) + this.width(), this.height());
-            ctx.fill();
-
-            ctx.beginPath();
-            ctx.fillStyle = "black";
-            ctx.textBaseline = "middle";
-            ctx.textAlign = "left";
-            ctx.font = "16px Monospace";
-            ctx.fillText(this.cell.name, this.maxX-mapOffset.x, this.minY + this.height() / 2 - mapOffset.y);
-            ctx.fill();
-        }
-    }
-    ctx.drawImage(this.icon, this.minX-mapOffset.x, this.minY-mapOffset.y, this.width(), this.height());
-    //if(discoveredArr.includes(locObj.formid)){
-    //    ctx.drawImage(icons.Check, this.minX, this.minY, this.width(), this.height());
-    //}
-}
-
 /*********************************
  * TOPBAR FUNCTIONS
  *  this is the "topbar" on the map canvas.
  *********************************/
 
 function initTopbar(){
-    function MapButton(x,y,width,height,text){
+    function MapButton(ordinal,y,height,text){
```

Comment:
This class, despite being named fairly generically "mapButton", is actually pretty specific to the topbar since I included the ordinal parameter.